sound/soc/sof/loader.c

```diff
@@ -192,6 +192,24 @@ static int ext_man_get_dbg_abi_info(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+static int ext_man_get_config_data(struct snd_sof_dev *sdev,
+				    const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_config_data *config_data =
+		container_of(hdr, struct sof_ext_man_config_data, hdr);
+	int i;
+
+	/* TODO: Currently CAVS_LPRO only supported */
+	for (i = 0; i < SOF_EXT_MAN_CONFIG_LAST_ELEM; i++)
```

Comment:
The number of elems should be extracted from the header no? Because you may or may not have an element

Reply:
This is good point but this is not dynamic. `elems` is predefined size of array, decided by SOF_EXT_MAN_CONFIG_LAST_ELEM. Once sof_ext_man_config_data is found, `elems` are fixed size. I think using switch/case is enough.
```
/* extended data config data */
struct sof_ext_man_config_data {
	struct sof_ext_man_elem_header hdr;

	struct sof_config_elem elems[SOF_EXT_MAN_CONFIG_LAST_ELEM];
} __packed;
```

With this log, I was testing with 5 elems, header size 8 + elems size 40 = 48 = 0x30.
```
[    2.668769] sof-audio-pci 0000:00:1f.3: found sof_ext_man header type 5 size 0x30
[    2.668772] sof-audio-pci 0000:00:1f.3: clk_config CAVS_LPRO: 1
```

Reply:
I think the most reliable way is using hdr size as @ranj063 adviced.
SOF_EXT_MAN_CONFIG_LAST_ELEM can be different on kernel and FW, imagine the situation when the value is higher on the kernel side